### PR TITLE
Handling API Rate limit - wait for required seconds and retry request

### PR DESCRIPTION
**Pull request by ganeshw71**
- Handled API Rate limit. It will wait for the required wait-time and retry the request

- Added proxy support for the script.

- Handled API Concurrency Limit. It will check the API concurrency limit first and then allow the number of threads to run. Also, if concurrency limit reached while calling API then thread will sleep for 30 seconds and will retry the request

### CODE REVIEW

**Review by pmgupte — CHANGES_REQUESTED** [review]
@ganeshwaghmare As discussed, lets also handle case of concurrency limit and add proxy support. We can then merge this PR.

File: hostdetection/multi_thread_hd.py [hunk 1]
```diff
@@ -55,6 +57,26 @@ def build_request(api_route, params):
     return urllib2.Request(api_route, data=data, headers=build_headers())
 # end of build_request
 
+def get_concurrency_limit(api_route):
+    '''
+    validate credentials and return API concurrency limit for the subscription
+    '''
+    try:
+        print "Calling %s " % api_route
+        req = build_request(api_route, {})
```
Comment (pmgupte) [inline hostdetection/multi_thread_hd.py]: @ganeshwaghmare can `call_api(api_route, params)` be used here?

**Review by pmgupte — CHANGES_REQUESTED** [review]
Requesting changes in `get_concurrency_limit(api_route)` method to use `call_api(api_route, params)` method.

File: hostdetection/multi_thread_hd.py (hunk 1 above)
Comment (ganeshw71) [reply hostdetection/multi_thread_hd.py]: Done with the changes.

**Review by pmgupte — APPROVED** [review]

**Pull request merged**